### ISSUE TITLE
Error 'Request entity too large' only with --reuse-values option

### ISSUE THREAD

**Issue by lictw** [helm/helm#10601]
Hi! The Chart normally installs/upgrades while `--reuse-values` option does not used, otherwise error occurs:
```
$ helm upgrade --reuse-values NAME .
Error: UPGRADE FAILED: create: failed to create: Request entity too large: limit is 3145728

$ helm get values -o yaml NAME > /tmp/values.yaml 
$ helm template -f /tmp/values.yaml . | wc -c
83394
$ helm upgrade -f /tmp/values.yaml NAME .
Release "NAME" has been upgraded. Happy Helming!
...
```
What's wrong here? Others Charts with similar size works fine in same environment.
```
$ helm template -f /tmp/values.yaml . | wc -c
69380
$ helm upgrade --reuse-values foo .
Release "foo" has been upgraded. Happy Helming!
```

Output of `helm version`:
```
version.BuildInfo{Version:"v3.7.2", GitCommit:"663a896f4a815053445eec4153677ddc24a0a361", GitTreeState:"clean", GoVersion:"go1.17.4"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"archive", BuildDate:"2021-03-25T19:02:48Z", GoVersion:"go1.16.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.8+k3s2", GitCommit:"be3c430985740eb6f8d16f977ee0e742aa63c530", GitTreeState:"clean", BuildDate:"2022-01-14T07:23:46Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): K3s

**Comment by lictw** [comment]
The same chart in other configuration (it's unicorn chart) also works fine, i think deals in size.
```
$ helm upgrade --reuse-values NAME .
Release "NAME" has been upgraded. Happy Helming!
$ helm template -f /tmp/values.yaml . | wc -c
73776
```

**Comment by yxxhero** [comment]
by the --debug option. show more message @lictw Thanks very much.

**Comment by lictw** [comment]
How about this? Sometime it does not pass to error.. And rarely it can pass normally, but same operation with -f values.yaml passes with 100% rate.
![image](https://user-images.githubusercontent.com/11904484/151226506-08e78ff0-bc34-4ac4-8f61-36bea4118f34.png)

**Comment by lictw** [comment]
Hi! There are some updates.

I found that sh.helm.release.v1.* have anomaly huge size and increases with every new release, for example one from the lasts has **1347872** bytes! I decoded it and saw many duplicates of values, so i started to debug. 

The chart is a big unicorn where some apps made as sub-chart and some creates in a loop, in both cases i use includes for common things. For part in a loop i made very strange thing called **$ctx**, goal was to be able to use same defines as i use in sub-charts, to achieve this i should create pseudo **$** in a loop and i don't fully understand what's going on here, but it works:
```
{{- $G := .Values.global }}
  {{- range $G.services }}
    {{- $chart := dict "Name" .name }}
    {{- $ctx := merge (dict "Values" .) (dict "Chart" $chart) (deepCopy $) }}
...
apiVersion: apps/v1
kind: Deployment
metadata:
  name: {{ .name }}
  labels:
    {{- include "labels" $ctx | indent 4 }}
  annotations:
    {{- include "annotations" $ctx | indent 4 }}
```
It's from template in main/parent chart and already i know for sure (via debug) that it's root of the problem and somehow it duplicates values of all sub-charts in itself and itself in values of all sub-charts and many, many and many times in a row. 

I will rewrite all at sub-charts (i wanted it before this happed), but anyway i want to understand what's wrong here, especially why size of every next sh.helm.release.v1.* more than prev., minor, about 150 bytes, but more.

_PS: Sorry for my English, i hope all main thoughts are clear.._
